base/Base/Base_Base_implementation.F90

```diff
@@ -3169,6 +3169,174 @@ end function ijsearch
+       grid_is_ok = .true.
+    endif
+
+    allocate(lons(npts),lats(npts))
+    if (present(lon) .and. present(lat)) then
+       lons = lon
+       lats = lat
+    else if (present(lonR8) .and. present(latR8)) then
+       lons = lonR8
+       lats = latR8
+    end if
+
+    !shift the conner away from Japan Fuji Mt.
+    lons = lons + Japan_shift
+
+    ! get xyz from sphere surface
+    allocate(xyz(3, npts), max_abs(npts))
+    xyz(1,:) = cos(lats)*cos(lons)
+    xyz(2,:) = cos(lats)*sin(lons)
+    xyz(3,:) = sin(lats)
+
+    ! project onto cube
+    max_abs = maxval(abs(xyz), dim=1)
+
+    xyz(1,:) = xyz(1,:)/max_abs
+    xyz(2,:) = xyz(2,:)/max_abs
+    xyz(3,:) = xyz(3,:)/max_abs
+
+    allocate(x, source=xyz(1,:))
+    allocate(y, source=xyz(2,:))
+    allocate(z, source=xyz(3,:))
+
+    II = -1
+    JJ = -1
+
+    ! Here the error and shift are used for edge points.
+    ! the edge points are assigned in the order of face 1,2,3,4,5,6
+    error = epsilon(1.0d0)
+    shift = 10.0d0*error
+
+    ! face = 1
+    where ( abs(x-1.0d0) <= error )
+       ! Four edges assigned to face 1
+       where (abs(y - 1.0d0) <=error) y = y-shift
+       where (abs(y + 1.0d0) <=error) y = y+shift
+       where (abs(z - 1.0d0) <=error) z = z-shift
+       where (abs(z + 1.0d0) <=error) z = z+shift
+       II = ceiling((atan(y/sqr2) + alpha)/dalpha)
+       JJ = ceiling((atan(z/sqr2) + alpha)/dalpha)
+    endwhere
+
+    ! face = 2
+    where (abs(y-1.0d0) <= error)
+       ! Three edges assigned to face 2
+       where (abs(x + 1.0d0) <=error) x = x+shift
+       where (abs(z - 1.0d0) <=error) z = z-shift
+       where (abs(z + 1.0d0) <=error) z = z+shift
+       II = ceiling((atan(-x/sqr2) + alpha)/dalpha)
+       JJ = ceiling((atan(z/sqr2) + alpha)/dalpha)
+       JJ = JJ + IM_WORLD
+    endwhere
+
+    ! face = 3
+    where (abs(z-1.0d0) <= error)
+       ! Two edges assigned to face 3
+       where (abs(y +  1.0d0) <=error) y = y+shift
+       where (abs(x +  1.0d0) <=error) x = x+shift
+       II = ceiling((atan(-x/sqr2) + alpha)/dalpha)
+       JJ = ceiling((atan(-y/sqr2) + alpha)/dalpha)
+       JJ = JJ + IM_WORLD*2
+    endwhere
+
+    ! face = 4
+    where (abs(x+1.0d0) <= error)
+       ! Two edges assigned to face 4
+       where (abs(y +  1.0d0) <=error) y = y+shift
+       where (abs(z +  1.0d0) <=error) z = z+shift
+       II = ceiling((atan(-z/sqr2) + alpha)/dalpha)
+       JJ = ceiling((atan(-y/sqr2) + alpha)/dalpha)
+       JJ = JJ + IM_WORLD*3
+    endwhere
+
+    ! face = 5
+    where (abs(y+1.0d0) <= error)
+       ! One edge assigned to face 5
+       where (abs(z +  1.0d0) <=error) z = z+shift
+       II = ceiling((atan(-z/sqr2) + alpha)/dalpha)
+       JJ = ceiling((atan(x/sqr2) + alpha)/dalpha)
+       JJ = JJ + IM_WORLD*4
+    endwhere
+
+    ! face = 6
+    where (abs(z+1.0d0) <= error)
+       II = ceiling((atan(y/sqr2) + alpha)/dalpha)
+       JJ = ceiling((atan(x/sqr2) + alpha)/dalpha)
+       JJ = JJ + IM_WORLD*5
```

Comment:
Do we really still need the fake JJ?    I would love to find a way to work instead with a proper `(I,J,face)` tuple.

Reply:
I am open to this. if we use (I, J, face), we need to change a little bit on MAPL_GridGetInterior